src/lib/y2network/routing_helpers.rb

```diff
@@ -0,0 +1,110 @@
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+require "y2network/routing_table"
+require "y2network/route"
+
+module Y2Network
+  module RoutingHelpers
+
+  private
+
+    MISSING_VALUE = "-".freeze
+    private_constant :MISSING_VALUE
+    DEFAULT_DEST = "default".freeze
+    private_constant :DEFAULT_DEST
+
+    # Find configured network interfaces
+    #
+    # Configured interfaces has a configuration (ifcfg file) assigned.
+    #
+    # @return [Array<Interface>] Detected interfaces
+    # @see Yast::NetworkInterfaces.Read
+    def find_interfaces
+      Yast::NetworkInterfaces.Read
+      # TODO: for the time being, we are just relying in the underlying stuff.
+      Yast::NetworkInterfaces.List("").map do |name|
+        Y2Network::Interface.new(name)
+      end
+    end
+
+    # Given an IP and a netmask, returns a valid IPAddr object
+    #
+    # @param ip_str      [String] IP address; {MISSING_VALUE} means that the IP is not defined
+    # @param netmask_str [String] Netmask; {MISSING_VALUE} means that no netmask was specified
+    # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+    def build_ip(ip_str, netmask_str = MISSING_VALUE)
+      return nil if ip_str == MISSING_VALUE
+
+      ip = IPAddr.new(ip_str)
+      netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+    end
+
+    # Build a route given a hash from the SCR agent
+    #
+    # @param hash [Hash] Hash from the `.routes` SCR agent
+    # @return Route
+    def build_route(hash)
+      # TODO: check whether the iface is configured in the system
+      iface = Interface.new(hash["device"])
+      # normalized SCR output contains either subnet mask or /<prefix length> under
+      # "netmask" key
+      # TODO: this should be improved in normalize_routes
+      mask = hash["netmask"] =~ /\/[0-9]+/ ? hash["netmask"][1..-1] : hash["netmask"]
+
+      Y2Network::Route.new(
+        to:        hash["destination"] != DEFAULT_DEST ? build_ip(hash["destination"], mask) : :default,
+        interface: iface,
+        gateway:   build_ip(hash["gateway"], MISSING_VALUE),
+        options:   hash["extrapara"]
+      )
+    end
+
+    # Returns a hash containing the route information to be imported into {Yast::Routing}
+    #
+    # @param route [Y2Network::Route]
+    # @return [Hash]
+    def route_to_hash(route)
+      hash =
+        if route.default?
+          { "destination" => DEFAULT_DEST, "netmask" => "-" }
+        else
+          { "destination" => route.to.to_s, "netmask" => netmask(route.to) }
+        end
+      hash.merge("options" => route.options) unless route.options.to_s.empty?
```

Comment:
`extrapara` was used in the past, so it will break the current AY schema.

Reply:
so change that schema?

Reply:
@imobachgs is working in a presenter, we should maintain it as it is, we could also export as "options" if decided but then we will need to import using both keys